crates/typst-library/src/text/quote.rs

```diff
@@ -0,0 +1,136 @@
+use super::{SmartquoteElem, SpaceElem, TextElem};
+use crate::{
+    layout::{BlockElem, HElem, PadElem, Spacing},
+    prelude::*,
+};
+
+/// Displays a quote alongside it's author.
+///
+/// # Example
+/// ```example
+/// Plato is often misquoted as the author of #quote[I know that I know
+/// nothing], however, this is a derivation form his orginal quote:
+/// #quote(block: true, author: [Plato])[
+///   ... ἔοικα γοῦν τούτου γε σμικρῷ τινι αὐτῷ τούτῳ σοφώτερος εἶναι, ὅτι
+///   ἃ μὴ οἶδα οὐδὲ οἴομαι εἰδέναι.
+/// ]
+/// #quote(
+///  block: true,
+///  author: [from the Henry Cary literal translation of 1897]
+/// )[
+///   ... I seem, then, in just this little thing to be wiser than this man at
+///   any rate, that what I do not know I do not think I know either.
+/// ]
+/// ```
+#[elem(Show)]
+pub struct QuoteElem {
+    /// Whether this is a block quote.
+    ///
+    /// ```example
+    /// #quote(author: [René Descartes])[cogito, ergo sum]
+    ///
+    /// #set quote(block: true)
+    /// #quote(author: [JFK])[Ich bin ein Berliner.]
+    /// ```
+    block: bool,
+
+    /// Whether double quotes should be added around this quote.
+    ///
+    /// - `{true}`: Wrap this quote in double quotes.
+    /// - `{false}`: Do not wrap this quote in double quotes.
+    /// - `{auto}`: Infer whether to wrap this quote in double quotes based on
+    ///   the `block` property. If `block` is `{false}` double quotes are
+    ///   auomatically added.
+    ///
+    /// ```example
+    /// #set text(lang: "de")
+    /// #quote[Ich bin ein Berliner.]
+    ///
+    /// #set smartquote(quotes: "«»")
+    /// #set quote(quotes: true)
+    /// #quote(block: true)[
+    ///   ... ἔοικα γοῦν τούτου γε σμικρῷ τινι αὐτῷ τούτῳ σοφώτερος εἶναι, ὅτι
+    ///   ἃ μὴ οἶδα οὐδὲ οἴομαι εἰδέναι.
```

Comment:
lang is still "de". that doesn't seem ideal.

Reply:
Yeah, this was more so to show that smartquote.quotes are used, I'll use a german quote instead.